apps/rush-lib/src/cli/utilities/ShrinkwrapFile.ts

```diff
@@ -128,18 +52,23 @@ export default class ShrinkwrapFile {
     }
   }
 
-  private constructor(shrinkwrapJson: IShrinkwrapJson) {
-    this._shrinkwrapJson = shrinkwrapJson;
+  /**
+   * Returns the list of temp projects defined in this file.
+   * Example: [ '@rush-temp/project1', '@rush-temp/project2' ]
+   */
+  public abstract getTempProjectNames(): ReadonlyArray<string>;
+
+  protected abstract getDependencyVersion(dependencyName: string, tempProjectName?: string): string | undefined;
 
-    // Normalize the data
-    if (!this._shrinkwrapJson.version) {
-      this._shrinkwrapJson.version = '';
-    }
-    if (!this._shrinkwrapJson.name) {
-      this._shrinkwrapJson.name = '';
-    }
-    if (!this._shrinkwrapJson.dependencies) {
-      this._shrinkwrapJson.dependencies = { };
+  protected _getTempProjectNames(dependencies: { [key: string]: {} } ): ReadonlyArray<string> {
```

Comment:
>_getTempProjectNames [](start = 12, length = 20)

Prefer move this one to NpmShrinkwrapFile class. If pnpm needs it in the future, move it at that time.